metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -2360,4 +2363,9 @@ void setNewNextWriteOffset(long newNextWriteOffset) {
     void handleUncleanBrokerShutdown(int brokerId, List<ApiMessageAndVersion> records) {
         replicationControl.handleBrokerUncleanShutdown(brokerId, records);
     }
+
+    void maybeTriggerMinIsrConfigUpdate(Optional<String> topicName) throws InterruptedException, ExecutionException {
+        appendWriteEvent("partitionUpdateForMinIsrChange", OptionalLong.empty(),
+            () -> replicationControl.getPartitionElrUpdatesForConfigChanges(topicName)).get();
+    }
```

Comment:
calling `.get()` on an appendWriteEvent doesn't look right to me. If I understand correctly, the `appendWriteEvents` are handled in the quorum controller event loop thread.

We would expect `replay()` to also be called in the event loop thread. so if we trigger an `appendWriteEvent` and block waiting for the result, it would always time out, since we are blocking the processing thread.

Reply:
Then we only need to call appendWriteEvent here? We don't have to wait for the replay().

Reply:
Got it, we basically only need to call the appendWriteEvents and do not wait for the replay().

Reply:
hmm, I think a better way to think about it is that we want to append the min ISR config update atomically with the partition change records. Only appending the partition change records once the config change is replayed is difficult to reason about and possibly incorrect.

Perhaps one example to look at is `ReplicationControlManager.handleBrokerFenced`. When a broker is fenced, we generate a broker registration change record along with the leaderAndIsr partition change records. I assume we want to follow a similar model with the topic configuration change events.

Reply:
Make sense, I have some misunderstanding about the controller events. Will update. Thanks!